Adding vote to force-pause players (instead of just moving them to spectators).

ravomavain opened the pull request:
This add the following config variables :
*sv_vote_pause: Allow voting to pause players (instead of moving to spectators) (default: 1)
*sv_vote_pause_time: The time (in seconds) players have to wait in pause when paused by vote (default: 60)
*sv_vote_pause_auto: Automatically unpaused players after the force-pause delay or not (default: 1)

And these functions :
- force_pause v?i : Force v to pause for i seconds
- force_unpause v : Force v to unpause

ravomavain committed:
Adding vote to force-pause players (instead of just moving them to spectators).

ravomavain commented:
Wait ! This need some fixes ;-)

Shahan commented:
Nice idea, it was suggested by GreYFoX, as I remember, but on ddrace.info.
I like your jail system and votes to put player to jail. Why don't you suggest it here? It's really useful!

ravomavain commented:
GreYFoX asked me on irc to do it like that, Jail is a bit too childish/noobish (which may be why you actually like it ^^) and it require you to modify every maps (and mapers don't like when we add a jail to there map). So we went to this idea of force-pause which is quiet the same but don't require map modification.
For now it's not really finished since I don't know if players in team can be force-paused by vote (since pause is disabled in team). For the moment it can pause any player even if they are in team, are moving,...

ravomavain committed:
Fixing some pause/team bugs :
* You can now use /pause in team.
* Team can't finish if a paused player have not finished.
* You can't join a team that have already started, even if all its members are paused.
Also adding pause specific message when players use /pause (instead of joined the game/spectators).

alandball commented:
Also, how about afk force-pause. this might be better than afk-kick, but will depend on server hosts and how evil they feel. And then spectator afk kick if too many in spectator, eg kick the longest afk spectator if spectators reach a certain amount after t minutes, to avoid kicking unnecessary tee incase tournament mode is on, this way tee wont get kicked upon map change.

Shahan commented:
" too childish/noobish (which may be why you actually like it ^^) "
What do you mean? It's insult? I meant jail not on map, i means freeze jail, just freeze for any time without porting somewhere.

"Also, how about afk force-pause. this might be better than afk-kick, but will depend on server hosts and how evil they feel. And then spectator afk kick if too many in spectator, eg kick the longest afk spectator if spectators reach a certain amount after t minutes, to avoid kicking unnecessary tee incase tournament mode is on, this way tee wont get kicked upon map change."
agreed

ravomavain commented:
Was not an insult, just that, like me, you seems to like features that are "just for fun" but that are not really wanted in official DDRace ^^. Btw freezing player on the map where they are is a bad idea because it may block other players. Force pausing them is a better idea.

For the afk force-pause it can be a good idea, but one thing at a time, we have first to fix teams and pause ^^ (almost done)

I'm closing this for now because I'm going to make a new pull request (proper and cleaner) ;-)

Shahan commented:
Hm. For fun.. I stopped making smth for fun half year ago. I even didn't unblock cheats..
I had only 1 funserver of 8 and you joined only funserver. I think that's why you think i do everything only for fun. I try not to cahnge admin stuff, i want make smth interesting for every player.
About dummies, i don't want pull request about adding it to of. ddrace because it's break main idea of DDRace:
"The target of the game is to reach the finish line. But in contrast to other races you can't reach the finish alone. DDRace requires teamwork! The success does not only depend on how good the single players are, but also how good they are in playing in a team."

Jail is nice idea, it can be used in other mod as interestiong addition. Yes, freeze much worse, pause players much better, agreed.

ravomavain commented:
What I mean by "just for fun" was that these features are not intended to be added in official DDRace and have been _coded_ just for the fun (because they are not really reused in other mods like you dummies). Anyway, no offense ;)
